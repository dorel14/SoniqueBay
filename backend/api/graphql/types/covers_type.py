# from strawchemy_init import strawchemy
# from ...models.covers_model import Cover
# import strawberry

# @strawberry.enum
# class CoverType:
#     ARTIST = "artist"
#     ALBUM = "album"
#     TRACK = "track"


# @strawchemy.order(Cover, include="all")
# class CoverOrder:
#     pass
# @strawchemy.filter(Cover, include="all")
# class CoverFilter:
#     pass
# @strawchemy.type(Cover, include="all", filter_input=CoverFilter, order_by=CoverOrder, override=True)
# class CoverGQL:
#     pass
# @strawchemy.create_input(Cover, exclude=["id", "created_at", "updated_at"])
# class CoverCreateInput:
#     pass