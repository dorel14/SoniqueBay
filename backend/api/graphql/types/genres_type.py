from __future__ import annotations
from typing import Annotated
from strawberry import auto
# Temporarily disable Strawchemy to avoid conflicts
# from backend.api.graphql.strawchemy_init import strawchemy
from backend.api.models.genres_model import Genre
import strawberry

@strawberry.type
class GenreType:
    id: int
    name: str
    musicbrainz_genreid: str | None