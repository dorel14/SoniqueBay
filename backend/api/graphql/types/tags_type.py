from __future__ import annotations
from typing import Annotated
from strawberry import auto
# Temporarily disable Strawchemy to avoid conflicts
# from backend.api.graphql.strawchemy_init import strawchemy
from backend.api.models.tags_model import GenreTag, MoodTag
import strawberry

@strawberry.type
class GenreTagType:
    id: int
    name: str

@strawberry.type
class MoodTagType:
    id: int
    name: str
