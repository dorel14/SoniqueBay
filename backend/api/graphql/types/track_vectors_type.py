from __future__ import annotations
from strawberry import auto
from backend.api.graphql.strawchemy_init import strawchemy
from backend.api.models.track_vectors_model import TrackVector

@strawchemy.order(TrackVector, include="all")
class TrackVectorOrderedType: ...
@strawchemy.filter(TrackVector, include="all")
class TrackVectorFilterType: ...
@strawchemy.type(TrackVector, include="all",filter_input=TrackVectorFilterType, order_by=TrackVectorOrderedType, override=True)
class TrackVectorType: ...
