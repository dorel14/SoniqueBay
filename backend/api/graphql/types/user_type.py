from ..graph_init import strawchemy
from ...models.user_model import User

@strawchemy.type(model=User, include=["id","username"])
class UserType():
    pass

@strawchemy.filter(model=User, include=["username"])
class UserFilter():
    pass

@strawchemy.order(model=User, include=["username"])
class UserOrderBy():
    pass