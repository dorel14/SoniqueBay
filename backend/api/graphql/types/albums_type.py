from __future__ import annotations
import strawberry
from strawberry import auto
from backend.api.models.albums_model import Album
from .covers_type import CoverType


@strawberry.type
class AlbumType:
    id: int
    title: str
    album_artist_id: int
    release_year: str | None
    musicbrainz_albumid: str | None
    covers: list[CoverType] = strawberry.field(default_factory=list)


@strawberry.input
class AlbumCreateInput:
    title: str
    album_artist_id: int
    release_year: str | None = None
    musicbrainz_albumid: str | None = None


@strawberry.input
class AlbumUpdateInput:
    id: int
    title: str | None = None
    album_artist_id: int | None = None
    release_year: str | None = None
    musicbrainz_albumid: str | None = None
