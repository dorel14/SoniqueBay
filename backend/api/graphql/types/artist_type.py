from __future__ import annotations
import strawberry
from strawberry import auto
from backend.api.models.artists_model import Artist
from backend.api.graphql.types.covers_type import CoverType


@strawberry.type
class ArtistType:
    id: int
    name: str
    musicbrainz_artistid: str | None
    covers: list[CoverType] = strawberry.field(default_factory=list)


@strawberry.input
class ArtistCreateInput:
    name: str
    musicbrainz_artistid: str | None = None


@strawberry.input
class ArtistUpdateInput:
    id: int
    name: str | None = None
    musicbrainz_artistid: str | None = None