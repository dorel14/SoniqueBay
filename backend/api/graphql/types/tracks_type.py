from __future__ import annotations
import strawberry
from strawberry import auto
from backend.api.models.tracks_model import Track
from backend.api.graphql.types.covers_type import CoverType
from backend.api.graphql.types.artist_type import ArtistType
from backend.api.graphql.types.albums_type import AlbumType


@strawberry.type
class TrackType:
    id: int
    title: str | None
    path: str
    track_artist_id: int
    album_id: int | None
    duration: float | None
    track_number: str | None
    disc_number: str | None
    year: str | None
    genre: str | None
    file_type: str | None
    bitrate: int | None
    featured_artists: str | None
    bpm: float | None
    key: str | None
    scale: str | None
    danceability: float | None
    mood_happy: float | None
    mood_aggressive: float | None
    mood_party: float | None
    mood_relaxed: float | None
    instrumental: float | None
    acoustic: float | None
    tonal: float | None
    camelot_key: str | None
    genre_main: str | None
    musicbrainz_id: str | None
    musicbrainz_albumid: str | None
    musicbrainz_artistid: str | None
    musicbrainz_albumartistid: str | None
    acoustid_fingerprint: str | None
    covers: list[CoverType] = strawberry.field(default_factory=list)


@strawberry.input
class TrackCreateInput:
    title: str | None = None
    path: str
    track_artist_id: int
    album_id: int | None = None
    duration: float | None = None
    track_number: str | None = None
    disc_number: str | None = None
    year: str | None = None
    genre: str | None = None
    file_type: str | None = None
    bitrate: int | None = None
    featured_artists: str | None = None
    bpm: float | None = None
    key: str | None = None
    scale: str | None = None
    danceability: float | None = None
    mood_happy: float | None = None
    mood_aggressive: float | None = None
    mood_party: float | None = None
    mood_relaxed: float | None = None
    instrumental: float | None = None
    acoustic: float | None = None
    tonal: float | None = None
    camelot_key: str | None = None
    genre_main: str | None = None
    musicbrainz_id: str | None = None
    musicbrainz_albumid: str | None = None
    musicbrainz_artistid: str | None = None
    musicbrainz_albumartistid: str | None = None
    acoustid_fingerprint: str | None = None


@strawberry.input
class TrackUpdateInput:
    id: int
    title: str | None = None
    path: str | None = None
    track_artist_id: int | None = None
    album_id: int | None = None
    duration: float | None = None
    track_number: str | None = None
    disc_number: str | None = None
    year: str | None = None
    genre: str | None = None
    file_type: str | None = None
    bitrate: int | None = None
    featured_artists: str | None = None
    bpm: float | None = None
    key: str | None = None
    scale: str | None = None
    danceability: float | None = None
    mood_happy: float | None = None
    mood_aggressive: float | None = None
    mood_party: float | None = None
    mood_relaxed: float | None = None
    instrumental: float | None = None
    acoustic: float | None = None
    tonal: float | None = None
    camelot_key: str | None = None
    genre_main: str | None = None
    musicbrainz_id: str | None = None
    musicbrainz_albumid: str | None = None
    musicbrainz_artistid: str | None = None
    musicbrainz_albumartistid: str | None = None
    acoustid_fingerprint: str | None = None