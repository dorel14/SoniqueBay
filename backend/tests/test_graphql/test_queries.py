"""Tests for GraphQL queries."""

import pytest
from strawberry.types import ExecutionResult

from .fixtures import (
    QUERY_GET_ARTIST,
    QUERY_LIST_ARTISTS,
    QUERY_GET_ALBUM,
    QUERY_LIST_ALBUMS,
    QUERY_GET_TRACK,
    QUERY_LIST_TRACKS,
    SAMPLE_ARTIST_INPUT,
)


class TestArtistQueries:
    """Tests for artist queries."""

    @pytest.mark.usefixtures("create_test_artist")
    def test_get_artist(self, execute_graphql, create_test_artist, snapshot):
        """Test retrieving a single artist by ID."""
        artist = create_test_artist(name="Test Artist")
        result: ExecutionResult = execute_graphql(
            QUERY_GET_ARTIST, {"id": artist.id}
        )
        assert not result.errors
        assert result.data
        snapshot.assert_match(result.data)

    @pytest.mark.usefixtures("create_test_artists")
    def test_list_artists(self, execute_graphql, create_test_artists, snapshot):
        """Test retrieving list of artists."""
        create_test_artists(count=2)
        result: ExecutionResult = execute_graphql(QUERY_LIST_ARTISTS)
        assert not result.errors
        assert result.data
        # Expect non-empty list
        assert len(result.data["artists"]) == 2
        snapshot.assert_match(result.data)

    def test_list_artists_empty(self, execute_graphql, snapshot):
        """Test list artists with empty DB."""
        result: ExecutionResult = execute_graphql(QUERY_LIST_ARTISTS)
        assert not result.errors
        assert result.data == {"artists": []}
        snapshot.assert_match(result.data)

    def test_get_artist_nonexistent(self, execute_graphql, snapshot):
        """Test getting non-existent artist."""
        result: ExecutionResult = execute_graphql(
            QUERY_GET_ARTIST, {"id": 999}
        )
        assert not result.errors
        assert result.data == {"artist": None}
        snapshot.assert_match(result.data)


class TestAlbumQueries:
    """Tests for album queries."""

    @pytest.mark.usefixtures("create_test_album")
    def test_get_album(self, execute_graphql, create_test_album, snapshot):
        """Test retrieving a single album by ID."""
        album = create_test_album(title="Test Album")
        result: ExecutionResult = execute_graphql(
            QUERY_GET_ALBUM, {"id": album.id}
        )
        assert not result.errors
        assert result.data
        snapshot.assert_match(result.data)

    @pytest.mark.usefixtures("create_test_albums")
    def test_list_albums(self, execute_graphql, create_test_albums, snapshot):
        """Test retrieving list of albums."""
        create_test_albums(count=2)
        result: ExecutionResult = execute_graphql(QUERY_LIST_ALBUMS)
        assert not result.errors
        assert result.data
        assert len(result.data["albums"]) == 2
        snapshot.assert_match(result.data)

    def test_get_album_nonexistent(self, execute_graphql, snapshot):
        """Test getting non-existent album."""
        result: ExecutionResult = execute_graphql(
            QUERY_GET_ALBUM, {"id": 999}
        )
        assert not result.errors
        assert result.data == {"album": None}
        snapshot.assert_match(result.data)


class TestTrackQueries:
    """Tests for track queries."""

    @pytest.mark.usefixtures("create_test_track")
    def test_get_track(self, execute_graphql, create_test_track, snapshot):
        """Test retrieving a single track by ID."""
        track = create_test_track(title="Test Track")
        result: ExecutionResult = execute_graphql(
            QUERY_GET_TRACK, {"id": track.id}
        )
        assert not result.errors
        assert result.data
        snapshot.assert_match(result.data)

    @pytest.mark.usefixtures("create_test_tracks")
    def test_list_tracks(self, execute_graphql, create_test_tracks, snapshot):
        """Test retrieving list of tracks."""
        create_test_tracks(count=2)
        result: ExecutionResult = execute_graphql(QUERY_LIST_TRACKS)
        assert not result.errors
        assert result.data
        assert len(result.data["tracks"]) == 2
        snapshot.assert_match(result.data)

    def test_get_track_nonexistent(self, execute_graphql, snapshot):
        """Test getting non-existent track."""
        result: ExecutionResult = execute_graphql(
            QUERY_GET_TRACK, {"id": 999}
        )
        assert not result.errors
        assert result.data == {"track": None}
        snapshot.assert_match(result.data)