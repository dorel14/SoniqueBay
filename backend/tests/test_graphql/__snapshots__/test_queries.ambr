# serializer version: 1
# name: TestAlbumQueries.test_get_album
  dict({
    'album': dict({
      'albumArtistId': 1,
      'id': 1,
      'musicbrainzAlbumid': 'test-mb-album-id',
      'releaseYear': '2023',
      'title': 'Test Album',
    }),
  })
# ---
# name: TestAlbumQueries.test_get_album_nonexistent
  dict({
    'album': None,
  })
# ---
# name: TestAlbumQueries.test_list_albums
  dict({
    'albums': list([
      dict({
        'albumArtistId': 1,
        'id': 1,
        'title': 'Test Album 0',
      }),
      dict({
        'albumArtistId': 1,
        'id': 2,
        'title': 'Test Album 1',
      }),
    ]),
  })
# ---
# name: TestArtistQueries.test_get_artist
  dict({
    'artist': dict({
      'id': 1,
      'musicbrainzArtistid': 'test-mb-id-93cb930b',
      'name': 'Test Artist',
    }),
  })
# ---
# name: TestArtistQueries.test_get_artist_nonexistent
  dict({
    'artist': None,
  })
# ---
# name: TestArtistQueries.test_list_artists
  dict({
    'artists': list([
      dict({
        'id': 1,
        'musicbrainzArtistid': 'test-mb-id-0',
        'name': 'Test Artist 0',
      }),
      dict({
        'id': 2,
        'musicbrainzArtistid': 'test-mb-id-1',
        'name': 'Test Artist 1',
      }),
    ]),
  })
# ---
# name: TestArtistQueries.test_list_artists_empty
  dict({
    'artists': list([
    ]),
  })
# ---
# name: TestTrackQueries.test_get_track
  dict({
    'track': dict({
      'albumId': 1,
      'bpm': 120.0,
      'genre': 'Rock',
      'id': 1,
      'key': 'C',
      'path': '/path/to/test.mp3',
      'scale': 'major',
      'title': 'Test Track',
      'trackArtistId': 1,
    }),
  })
# ---
# name: TestTrackQueries.test_get_track_nonexistent
  dict({
    'track': None,
  })
# ---
# name: TestTrackQueries.test_list_tracks
  dict({
    'tracks': list([
      dict({
        'id': 1,
        'path': '/path/to/test0.mp3',
        'title': 'Test Track 0',
        'trackArtistId': 1,
      }),
      dict({
        'id': 2,
        'path': '/path/to/test1.mp3',
        'title': 'Test Track 1',
        'trackArtistId': 1,
      }),
    ]),
  })
# ---
